Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -603,6 +603,78 @@ public class GraphTraverser: GraphTraversing {
         )
     }
 
+    public func targetsWithExternalDependencies() -> Set<GraphTarget> {
+        Set(graph.dependencies.compactMap { fromDependency, toDependencies in
+            let fromNonExternalTarget: GraphTarget?
+            switch fromDependency {
+            case let .target(targetName, projectPath):
+                let project = graph.projects[projectPath]!
+                if project.isExternal {
+                    fromNonExternalTarget = nil
+                } else {
+                    let target = graph.targets[projectPath]![targetName]!
+                    fromNonExternalTarget = GraphTarget(path: projectPath, target: target, project: project)
+                }
+            case .bundle, .framework, .library, .packageProduct, .sdk, .xcframework:
+                fromNonExternalTarget = nil
+            }
+            guard fromNonExternalTarget != nil else { return nil }
+
+            let dependsOnExternalDependency = toDependencies.first { toDependency in
+                switch toDependency {
+                case let .target(_, projectPath):
+                    graph.projects[projectPath]!.isExternal
+                case .bundle, .framework, .library, .packageProduct, .sdk, .xcframework:
+                    false
+                }
+            } != nil
+
+            return dependsOnExternalDependency ? fromNonExternalTarget : nil
+        })
+    }
+
+    public func externalTargets() -> Set<GraphTarget> {
```

Comment:
I'd name this `allExternalTargets` as it's very similar to `allInternalTargets`. I'd also take a look if we can share some logic between the two.

Reply:
I'll rename it.